tools/clang/include/clang/SPIRV/SpirvType.h

```diff
@@ -240,17 +246,28 @@ class RuntimeArrayType : public SpirvType {
     return t->getKind() == TK_RuntimeArray;
   }
 
-  const SpirvType *getElementType() const { return elementType; }
+  const SpirvType *getElementType() const {
+    return llvm::cast<SpirvType>(elementType);
+  }
 
 private:
   const SpirvType *elementType;
 };
 
 class StructType : public SpirvType {
 public:
-  StructType(llvm::ArrayRef<const SpirvType *> memberTypes,
-             llvm::StringRef name, llvm::ArrayRef<llvm::StringRef> memberNames,
-             bool isReadOnly);
+  enum class InterfaceType : uint32_t {
+    NonInterface = 0,
```

Comment:
I'd suggest to use more clear names as better documentation. And we should prefer Vulkan terms if possible. UBO/SSBO is essentially OpenGL and GLSL terms.

What about `InternalStorage`, `StorageBuffer`, `UniformBuffer`?